flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -468,12 +474,12 @@ private File getNextStoragePath() {
 
     @Override
     public CompletedCheckpointStorageLocation resolveCheckpoint(String pointer) throws IOException {
-        return checkpointStreamBackend.resolveCheckpoint(pointer);
+        return ((CheckpointStorage) checkpointStreamBackend).resolveCheckpoint(pointer);
```

Comment:
It seems possible to change the field type to `CheckpointStorage` and remove these casts.